packages/sign_in_with_apple/lib/src/exceptions.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart';
+
+/// A more specific [PlatformException] which describes any potential native errors that occur within the Sign in with Apple plugin.
+///
+/// Implementations:
+/// - [SignInWithAppleNotSupportedException]
+/// - [SignInWithAppleAuthorizationError]
+/// - [SignInWithAppleAuthorizationError]
+@immutable
+abstract class SignInWithAppleException implements Exception {
+  factory SignInWithAppleException.fromPlatformException(
+    PlatformException exception,
+  ) {
+    switch (exception.code) {
+      case 'not-supported':
+        return SignInWithAppleNotSupportedException(
+          message: exception.message,
+        );
+
+      /// Exceptions which indicate an [SignInWithAppleAuthorizationError]
+      case 'authorization-error/unknown':
+        return SignInWithAppleAuthorizationError(
```

Comment:
These should all be `Exception`s on the Dart side IMHO (https://groups.google.com/a/dartlang.org/forum/#!msg/misc/lx9CXiV3o30/s5l_PwpHUGAJ etc.), as they are expected to be caught and the programmer couldn't have prevented them.

Reply:
Jup, renamed 👍